src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.jsx

```diff
@@ -0,0 +1,86 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {t} from 'app/locale';
+import IndicatorStore from 'app/stores/indicatorStore';
+
+export default class AddIntegration extends React.Component {
+  static propTypes = {
+    children: PropTypes.func.isRequired,
+    provider: PropTypes.object.isRequired,
+    onInstall: PropTypes.func.isRequired,
+    reinstallId: PropTypes.string,
+  };
+
+  componentDidMount() {
+    this.dialog = null;
+    window.addEventListener('message', this.didReceiveMessage);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('message', this.didReceiveMessage);
+    this.dialog && this.dialog.close();
+  }
+
+  computeCenteredWindow(width, height) {
+    const screenLeft = window.screenLeft != undefined ? window.screenLeft : screen.left;
+
+    const screenTop = window.screenTop != undefined ? window.screenTop : screen.top;
+
+    const innerWidth = window.innerWidth
+      ? window.innerWidth
+      : document.documentElement.clientWidth
+        ? document.documentElement.clientWidth
+        : screen.width;
+
+    const innerHeight = window.innerHeight
+      ? window.innerHeight
+      : document.documentElement.clientHeight
+        ? document.documentElement.clientHeight
+        : screen.height;
+
+    const left = innerWidth / 2 - width / 2 + screenLeft;
+    const top = innerHeight / 2 - height / 2 + screenTop;
+
+    return {left, top};
+  }
+
+  openDialog = () => {
+    const name = 'sentryAddIntegration';
+    const {url, width, height} = this.props.provider.setupDialog;
+    const {reinstallId} = this.props;
+    const {left, top} = this.computeCenteredWindow(width, height);
+
+    const installUrl = reinstallId ? url + `?reinstall_id=${reinstallId}` : url;
+
+    const opts = `scrollbars=yes,width=${width},height=${height},top=${top},left=${left}`;
+
+    this.dialog = window.open(installUrl, name, opts);
+    this.dialog.focus();
+  };
+
+  didReceiveMessage = message => {
+    if (message.origin !== document.origin) {
+      return;
+    }
+
+    if (message.source !== this.dialog) {
+      return;
+    }
+
+    const {success, data} = message.data;
+    this.dialog = null;
+
+    if (!success) {
+      IndicatorStore.addError(data.error);
+      return;
+    }
+
+    this.props.onInstall(data);
+    IndicatorStore.addSuccess(t(`${this.props.provider.name} added`));
+  };
+
+  render() {
+    return <React.Fragment>{this.props.children(this.openDialog)}</React.Fragment>;
```

Comment:
I actually don't think you need the fragment

Reply:
I think React complains otherwise. Might be misremembering, though.

Reply:
I *think* this should be okay to just `return this.props.children(this.openDialog)`